tokio/src/sync/tests/loom_watch.rs

```diff
@@ -34,3 +36,32 @@ fn smoke() {
         th.join().unwrap();
     })
 }
+
+#[test]
+fn wait_for_test() {
+    loom::model(move || {
+        let (tx, mut rx) = watch::channel(false);
+
+        let tx_arc = Arc::new(tx);
+        let tx1 = tx_arc.clone();
+        let tx2 = tx_arc.clone();
+
+        let th1 = thread::spawn(move || {
+            for _ in 0..10 {
+                tx1.send(false).unwrap();
+                std::thread::sleep(Duration::from_millis(10));
+            }
```

Comment:
Please reduce the loop to 3 iterations. Also, you can use `send_modify` here to avoid unsetting the boolean when the other thread has set it to `true`.

Reply:
understood, updating